src/extension/providers/dart_reference_provider.ts

```diff
@@ -33,11 +34,16 @@ export class DartReferenceProvider implements ReferenceProvider, DefinitionProvi
 	}
 
 	public async provideDefinition(document: TextDocument, position: Position, token: CancellationToken): Promise<DefinitionLink[] | undefined> {
-		const resp = await this.analyzer.analysisGetNavigation({
-			file: fsPath(document.uri),
-			length: 0,
-			offset: document.offsetAt(position),
-		});
+		let resp1 = this.fileTracker.getNavigationTargets(fsPath(document.uri), document.offsetAt(position));
+		if (!resp1) {
+			resp1 = await this.analyzer.analysisGetNavigation({
+				file: fsPath(document.uri),
+				length: 0,
+				offset: document.offsetAt(position),
+			});
+		}
+		if (!resp1) return undefined;
```

Comment:
Is it possible for this to be undefined/null at this point? The original code seems to assume `analysisGetNavigation` can't return undefined, so I think it should still be the case after the above has run?

Reply:
Same as above, was just to satisfy the type system. Feel free to remove, you'll just have to add a couple of `1`s in the code below.

Reply:
I'd prefer not to have the `!` so this is fine - I just thought TypeScript was a little cleverer and wouldn't have reported it 🙄